tests/integration/devfile/cmd_devfile_create_test.go

```diff
@@ -87,398 +81,274 @@ var _ = Describe("odo devfile create command tests", func() {
 
-		Context("Testing Create for OpenShift specific scenarios", func() {
-			JustBeforeEach(func() {
-				if os.Getenv("KUBERNETES") == "true" {
-					Skip("This is a OpenShift specific scenario, skipping")
-				}
-			})
-
-			It("should fail when we create the devfile", func() {
-				helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", devfile), filepath.Join(commonVar.Context, devfile))
-				helper.Cmd("odo", "create", "nodejs").ShouldPass()
-				output := helper.Cmd("odo", "create", "nodejs").ShouldFail().Err()
-				Expect(output).To(ContainSubstring("this directory already contains a component"))
-			})
+	Context("odo create is executed with the --now flag", func() {
+		BeforeEach(func() {
+			helper.CopyExample(filepath.Join("source", "nodejs"), commonVar.Context)
 		})
 
-		Context("When devfile exists not in user's working directory and user specify the devfile path via --devfile", func() {
-			JustBeforeEach(func() {
-				newContext := path.Join(commonVar.Context, "newContext")
-				devfilePath = filepath.Join(newContext, devfile)
-				helper.MakeDir(newContext)
-				helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", devfile), devfilePath)
-			})
-
-			It("should successfully create the devfile component with valid file system path", func() {
-				helper.Cmd("odo", "create", "nodejs", "--devfile", devfilePath).ShouldPass()
-			})
-
-			It("should successfully create the devfile component with valid specifies URL path", func() {
-				helper.Cmd("odo", "create", "nodejs", "--devfile", "https://raw.githubusercontent.com/odo-devfiles/registry/master/devfiles/nodejs/devfile.yaml").ShouldPass()
-			})
-
-			It("should fail to create the devfile component with invalid file system path", func() {
-				helper.Cmd("odo", "create", "nodejs", "--devfile", "#@!").ShouldFail()
-			})
-
-			It("should fail to create the devfile component with invalid URL path", func() {
-				helper.Cmd("odo", "create", "nodejs", "--devfile", "://www.example.com/").ShouldFail()
-			})
-
-			It("should fail to create the devfile component with more than 1 arguments are passed in", func() {
-				helper.Cmd("odo", "create", "nodejs", "nodejs", "--devfile", devfilePath).ShouldFail()
-			})
-
-			It("should fail to create the devfile component with --registry specified", func() {
-				helper.Cmd("odo", "create", "nodejs", "--devfile", devfilePath, "--registry", "DefaultDevfileRegistry").ShouldFail()
-			})
+		It("checks that odo push works with a devfile with now flag", func() {
+			output := helper.Cmd("odo", "create", "nodejs", "--now").ShouldPass().Out()
+			Expect(output).To(ContainSubstring("Changes successfully pushed to component"))
 		})
 	})
 
-	Context("When executing odo create with devfile component and --starter flag", func() {
-		JustBeforeEach(func() {
+	Context("odo create is executed with the --starter flag", func() {
+		BeforeEach(func() {
 			contextDevfile = helper.CreateNewContext()
 			helper.Chdir(contextDevfile)
```

Comment:
Why are we moving to a different context? If necessary, can we express it in the Context or in a When?